src/basic/log.c

```diff
@@ -219,6 +220,30 @@ static int log_open_journal(void) {
         return r;
 }
 
+static bool stderr_is_journal(void) {
+        _cleanup_free_ char *f = NULL, *s = NULL;
+        const char *e;
+        uint64_t dev, ino;
+        struct stat st;
+
+        e = getenv("JOURNAL_STREAM");
+        if (!e)
+                return false;
+
+        if (extract_many_words(&e, ":", EXTRACT_DONT_COALESCE_SEPARATORS, &f, &s, NULL) < 2)
```

Comment:
hmm, to fully validate the input string you'd have to check `isempty(e)` after calling extract_many_words() to check that there's no trailing rubbish (think: a string line `47:11:8:15` is passed...)

I'd probably just use extract_first_word() to extract the first word, and then parse that to an integer and parse whatever remains as another integer (as my original code snippet suggested)